examples/lighting-app/nrf5/main/LightingCLI.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      The CLI interface of CHIP Lighting App
+ */
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include <lib/shell/shell.h>
+
+#include <lib/core/CHIPCore.h>
+#include <lib/support/Base64.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/RandUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/logging/CHIPLogging.h>
+
+#include <ChipShellCollection.h>
+
+#include <AppEvent.h>
+#include <AppTask.h>
+#include <LightingManager.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::Shell;
+
+namespace {
+
+enum CLIEvents
+{
+    kCLIEvent_LightOn = 0,
+    kCLIEvent_LightOff,
+    kCLIEvent_LightToggle,
+
+    kCLIEvent_FactoryReset,
+};
+
+void CLIAppEventHandler(AppEvent * event)
+{
+    switch (event->CLIEvent.Event)
+    {
+    case kCLIEvent_LightOn:
+        LightingMgr().InitiateAction(LightingManager::ON_ACTION);
+        break;
+    case kCLIEvent_LightOff:
+        LightingMgr().InitiateAction(LightingManager::OFF_ACTION);
+        break;
+    case kCLIEvent_LightToggle: {
+        LightingManager::Action_t action;
+        if (LightingMgr().IsTurnedOn())
+        {
+            action = LightingManager::OFF_ACTION;
+        }
+        else
+        {
+            action = LightingManager::ON_ACTION;
+        }
+        LightingMgr().InitiateAction(action);
+        break;
+    }
+    case kCLIEvent_FactoryReset:
+        ConfigurationMgr().InitiateFactoryReset();
+        return;
```

Comment:
Seems odd to me that some of these branches will use `break;` while others use `return;`. Would recommend just switching them all to `break;`

Reply:
Fixed, thanks for pointing out